/qbo_listen:
  ros__parameters:
    audio_device_name: "XVF3800"  # partie du nom du device audio d'entrée (ex: "Jabra", "ALSA", "USB", etc.)
    sample_rate: 16000
    audio_channels_opened: 2 # nombre de canaux effectivement ouverts sur le device (ex: Jabra envoie du stéréo même si on veut du mono, donc on ouvre 2 canaux et on sélectionne ensuite)
    asr_channel_index: 1 # indice du canal à utiliser pour l'ASR (ex: Jabra envoie du stéréo même si on veut du mono, donc on ouvre 2 canaux et on sélectionne ensuite)

    frame_ms: 20 # durée d'une frame audio en ms (10 ou 20ms conseillé)
    vad_enabled: true # active ou désactive la détection de voix (VAD)
    vad_mode: 2 # 0-3, plus c'est élevé, plus c'est strict (moins de faux positifs, mais plus de faux négatifs)
    silence_frames_end: 20 # 400ms de silence pour considérer la fin d'une parole
    min_utt_ms: 400 # durée minimale d'une parole (pour éviter les bruits courts)

    # anti tampon musique/bruit
    max_utt_ms: 4000 # durée maximale d'une parole (pour éviter les bruits longs, comme la musique ou le bruit de fond)
    min_speech_ratio: 0.50 # ratio minimum de frames de parole dans une utterance pour la valider (pour éviter les bruits longs avec peu de parole, comme la musique ou le bruit de fond)

    # lien TTS
    tts_ignore_ms: 200 # durée pendant laquelle on ignore les entrées audio après une sortie TTS (pour éviter de capter la propre voix du robot)

    # Whisper
    system_lang: "fr" #langue principale du système, pour aider
    whisper_model: "small" #taille du modèle Whisper à utiliser (tiny, base, small, medium, large)
    whisper_device: "cuda" #device pour faire tourner Whisper (cpu ou cuda)

    # confidence gating
    min_confidence: 0.5 #seuil de confiance minimum pour accepter une transcription (entre 0 et 1)
